agent/config/runtime.go

```diff
@@ -1350,14 +1350,16 @@ type RuntimeConfig struct {
 	// todo(fs): since they are standardized by IANA.
 	//
 	// hcl: tls_cipher_suites = []string
-	TLSCipherSuites []uint16
+	// TODO: is using types possible here or does this need to be a []string?
+	TLSCipherSuites []types.TLSCipherSuite
```

Comment:
How early in the config parsing pipeline is it possible to start using strongly-typed uint16 and string instead of passing around and storing raw config values?

Reply:
Ended up pushing validation all the way out to this layer.